### ISSUE TITLE
Memory Leak

### ISSUE THREAD

**Issue by malhal** [alin19xx/Loadable-View-ViewModel#1]
```
LoadableView(viewModel: ContentViewModel()) {
```
I noticed every time `body` is called another `ContentViewModel` object is init on the memory heap and lost, can we change `ContentViewModel` to a struct and pass in a binding to `LoadableView` instead?

**Comment by alin19xx** [comment]
You’re totally right, creating ContentViewModel directly in the body causes it to be reinitialized every time the view updates, which isn’t ideal. I’ve updated the code to use StateObject instead. This way, the ContentViewModel is only created once and keeps its state properly.

**Comment by malhal** [comment]
Next step is to remove the ContentViewModel class completely since ContentView already is a view model and since it is a struct has less chance of consistency errors compared to a class.
